concordium-contracts-common/src/impls.rs

```diff
@@ -281,6 +282,95 @@ impl Deserial for AccountBalance {
     }
 }
 
+impl<Kind> Serial for Threshold<Kind> {
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> { self.threshold.serial(out) }
+}
+
+impl<Kind> Deserial for Threshold<Kind> {
+    fn deserial<R: Read>(source: &mut R) -> ParseResult<Self> {
+        let threshold: u8 = u8::deserial(source)?;
+        if threshold == 0 {
+            Err(ParseError {})
+        } else {
+            Ok(Self {
+                threshold,
+                kind: marker::PhantomData,
+            })
+        }
+    }
+}
+
+impl<Kind> TryFrom<u8> for Threshold<Kind> {
+    type Error = ZeroSignatureThreshold;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        if value == 0 {
+            Err(ZeroSignatureThreshold)
+        } else {
+            Ok(Threshold {
+                threshold: value,
+                kind:      marker::PhantomData,
+            })
+        }
+    }
+}
+
+impl<Kind> Clone for Threshold<Kind> {
+    fn clone(&self) -> Self {
+        Self {
+            threshold: self.threshold,
+            kind:      marker::PhantomData,
+        }
+    }
+}
+
+impl<Kind> Copy for Threshold<Kind> {}
```

Comment:
Will this be implemented?

Reply:
This is a special marker trait, this implements it, there are no additional member functions to implement. I know it looks a bit odd though. https://doc.rust-lang.org/std/marker/trait.Copy.html

It's functionality is the same as `.clone`, except implementing this tells the compiler to do it implicitly.